src/sys/stat.rs

```diff
@@ -122,3 +122,88 @@ pub fn fstatat<P: ?Sized + NixPath>(dirfd: RawFd, pathname: &P, f: AtFlags) -> R
     Ok(dst)
 }
 
+/// The `chmod` function shall change `Mode::S_ISUID`, `Mode::S_ISGID`, `Mode::S_ISVTX`,
+/// and the file permission bits of the file named by the pathname
+/// pointed to by the `path` argument to the corresponding bits
+/// in the `mode` argument.
+///
+/// The application shall ensure that the effective user ID of the process
+/// matches the owner of the file or the process has appropriate privileges
+/// in order to do this.
+///
+/// If the calling process does not have appropriate privileges,
+/// and if the group ID of the file does not match the effective group ID
+/// or one of the supplementary group IDs and if the file is a regular file,
+/// bit `Mode::S_ISGID` (set-group-ID on execution) in the file's mode shall be cleared
+/// upon successful return from `chmod`.
+///
+/// Additional implementation-defined restrictions may cause
+/// the `Mode::S_ISUID` and `Mode::S_ISGID` bits in `mode` to be ignored.
+///
+/// Upon successful completion, `chmod` shall mark for update the last file
+/// status change timestamp of the file.
+///
+/// # References
+///
+/// See also [chmod(2)](http://pubs.opengroup.org/onlinepubs/9699919799/functions/chmod.html).
+pub fn chmod<P: ?Sized + NixPath>(path: &P, mode: Mode) -> Result<()> {
+    let res = try!(path.with_nix_path(|cstr| unsafe {
```

Comment:
You can replace `try!` with `?`.  See https://blog.rust-lang.org/2016/11/10/Rust-1.13.html for a description of how it works.